deck-core/src/main/kotlin/com/deck/core/entity/Channel.kt

```diff
@@ -8,8 +8,8 @@ import java.util.*
 
 public interface Channel : Entity {
     public val id: UUID
-    public val name: String
-    public val description: String
+    public val name: String?
+    public val description: String?
```

Comment:
Here it's up to you, if we either accept undefined descriptions as null or as blank strings.

Reply:
I think it would be better to move these properties to TeamChannel and leave them, at least the non-null `name`. What do you think?

Reply:
Since in TeamChannelCategory and PrivateChannel there is no `description`

Reply:
Yes, that's true. But both of them have names, so I don't understand why it's nullable here. By the way, does `TeamChannelCategory` represent a guilded category? If so, then it shouldn't implement `Channel`, don't you agree?

Reply:
I agree, incidentally I'm not reminded if `ChannelPrivate` have a `name`

Reply:
Yes, it has a name, but lacks an avatar or description.

Reply:
Understood!